### ISSUE TITLE
Following the "Getting started" documentation doesn't lead to the expected result

### ISSUE THREAD

**Issue by Duckwhale** [authelia/authelia#1864]
I've never used Authelia before and was simply trying to check out the demo. The setup script seems to have run successfully and the containers are ``up``. However, accessing either of the three subdomains described in the "Getting started" guide simply results in a ``404: Not Found`` . This is after the faux SSL warning from my browser, so it appears the host/port is indeed correct.

Alas, I'm not sure what should have happened instead... but... surely this isn't the desired result? As a dumb user who has no idea how Authelia works and wants to give it a try, I would expect to see something (anything, really) after the script tells me to check out the redirected``example.com`` URLs.

PS: This issue isn't just to ask what might've gone wrong, but also a suggestion to improve the documentation:

* If there is supposed to be some visible UI, add a paragraph with a screenshot to tell people what they should be seeing
* If there's any other way to verify the setup went through successfully, e.g., by checking the API endpoint is available or similar, that should also be noted
* If the expected outcome is "invisible", this is not ideal, but should also be noted
* If additional steps need to be taken before visible results manifest, all of them should be listed

Rationale:

The tutorial is the most critical part of any project; it'll be read by virtually every new user and most of them will be completely clueless... just like me :D It should work immediately and instructions should be provided that lead to the desired outcome, as well provisions included for any part where things could go wrong.

If it doesn't, lots of people will leave and find another project that "just works" (TM), never to be seen again (and obviously never supporting this one in any way), or they might open lazy support tickets, putting additional strain on the development team.

**Comment by nightah** [comment]
So I've noticed that I can reproduce the intial `404: Not found` behaviour, though this seems to be due to Traefik taking some time to finish starting up (~30 secs) after this everything works as normal. Can you please see if this is the same behaviour you're experiencing?

We could add a sleep or perform a little more logic in the setup script before telling a user they can visit the respective examples.

I'll see if I can figure out which version of Traefik or what specifically seems to be causing this issue.

**Comment by nightah** [comment]
Okay so it looks like the reason for the delay is Traefik generating a default certificate:
```
time="2021-03-30T03:08:51Z" level=debug msg="No default certificate, generating one"
```

I'm trying to determine why it's doing this even though we've specified the pre-generated certificates to be loaded into the default store. Either something has changed with Traefik configuration syntax or there's a bug upstream.

Having said that again this doesn't prevent the demo from working, there's just a delay while it generates that default certificate before a user can access the pages.

**Comment by Duckwhale** [comment]
> So I've noticed that I can reproduce the intial `404: Not found` behaviour, though this seems to be due to Traefik taking some time to finish starting up (~30 secs) after this everything works as normal. Can you please see if this is the same behaviour you're experiencing?

I'm not sure if this is the issue, but I did try to run ``setup.sh`` again after pulling in the latest changes and it appears to display the HTTP request details under ``https://public.example.com`` immediately now.  Is this the correct outcome?

I still think the expected results should be noted in the documentation, if only to make sure users can recognize if something didn't work as intended. If it takes some time to generate the certificate, perhaps amending the text to tell users to wait for a bit and try again if it didn't work immediately might also be helpful?

I can create a PR to improve the "Getting started" intro if you agree this would be useful. It's not a big deal, but making sure there is no doubt at all as to whether things worked (or when they didn't) can only improve the experience for new users.

**Comment by nightah** [comment]
The certificate was a red herring, the actual issue was the the docker health check was preventing from Traefik adding the routes for the Authelia container. This defaulted to 30 seconds each time, so the fix in #1871 removed that.

The setup script is quite simplistic and should technically be run on a fresh clone. If you tried just updating the repo after you had already run it once and run it again, yes it would have displayed the message quite quickly, but keep in mind that it also would not have updated the username and password (as this only currently works on a fresh clone).

More than happy for you to provide a PR to improve the "Getting started" section, that would probably help with some minor improvements to the setup script.

I think now that the issue has been identified and fixed for the setup script that new users would be unlikely to run into the same scenario you did. You can also test this by trying to run with a fresh clone. However I don't see how more information would hurt us in this scenario.

**Comment by Duckwhale** [comment]
> If you tried just updating the repo after you had already run it once and run it again, yes it would have displayed the message quite quickly, but keep in mind that it also would not have updated the username and password (as this only currently works on a fresh clone).

Yeah, I actually noticed that myself. I had a typo in my password and tried to run the script again to overwrite it, which didn't work. I had to replace it manually, by running the command that the script uses and then copying the output into the yml file.

Shouldn't running the setup again overwrite the database (yml file) so that it can be run multiple times if need be? Not sure how this would interact with the other setup stuff (docker/traefik) as I know nothing about that...
